src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```
@@ -0,0 +1,10 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    public void handle(FruitTransaction transaction, Map<String, Integer> stock) {
```

This handler does not validate its parameters. Per the task guidance, services and handlers should validate inputs and throw informative RuntimeException on invalid inputs (null parameters). Add checks such as `if (transaction == null) throw new RuntimeException("Transaction is null in PurchaseOperation.handle")` and `if (stock == null) throw new RuntimeException("Stock map is null in PurchaseOperation.handle")`. This prevents surprising NPEs and makes failures informative.